### PR TITLE
Adding basic caching example

### PR DESCRIPTION
**Pull request by bfosberry**

### CODE REVIEW

File: 17.caching/before.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+docker run -d --cidfile ./registry.cid -p 5000:5000 registry
+
```
Comment (beanieboi) [inline 17.caching/before.sh]: :scissors: newline

File: 17.caching/codeship-steps.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- service: demo
+  command: ls -l
+- service: demo
+  type: push
+  name: docker_push
+  image_name: localhost:5000/myapp
+  registry: localhost:5000
+  #encrypted_dockercfg_path: ./.dockercfg.encrypted
```
Comment (beanieboi) [inline 17.caching/codeship-steps.yml]: what about this comment?

File: 17.caching/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# base on latest ubuntu base image
+FROM ubuntu
+
+#do something that takes a long time
```
Comment (beanieboi) [inline 17.caching/Dockerfile]: please add a space after the `#` for consistency

File: 17.caching/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+RUN sleep 10s
+
+#do the thing
```
Comment (beanieboi) [inline 17.caching/Dockerfile]: please add a space after the # for consistency

File: 17.caching/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+```
+$ ./before.sh
+$ time jet steps --push --branch=test  # ~2mins
+$ time jet steps --push --branch=test  # ~20s due to caching plus no new images to push
+$ time jet steps --push --branch=test  # ~20s
+$ ./after.sh
+```
+
+You can switch off the `cached` flag in the codeship-services file to see what difference it makes.
+
```
Comment (beanieboi) [inline 17.caching/README.md]: :scissors: newline

File: 17.caching/codeship-steps.yml (hunk 2 above)
Comment (bfosberry) [inline 17.caching/codeship-steps.yml]: I can drop this now I think

**Pull request merged**